### PR TITLE
Sender now support sub-second period and delay

### PR DESCRIPTION
**Pull request by lionelvillard**
# Changes

- :gift: DELAY and PERIOD now supports Golang duration format. If the last unit is not specified, add 's' to the duration string.
-
-

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Comment by benmoss** [comment]
Why don't we just allow a time.Duration string?

**Comment by lionelvillard** [comment]
That's fine by me. We can even reuse DELAY and PERIOD and make `s` the default unit, for backward-compatible reason. WDYT?

**Comment by benmoss** [comment]
That sounds perfect, I was thinking we'd have to introduce a new field for compatibility, but you're right we can just default to seconds if not specified

**Comment by lionelvillard** [comment]
@benmoss can you TAL?

File: pkg/eventshub/sender/sender.go [hunk 1]
```diff
@@ -117,8 +118,17 @@ func Start(ctx context.Context, logs *eventshub.EventLogs) error {
 
 	logging.FromContext(ctx).Infof("Sender environment configuration: %+v", env)
 
-	period := time.Duration(env.Period) * time.Second
-	delay := time.Duration(env.Delay) * time.Second
+	p, err := time.ParseDuration(durationWithUnit(env.Period))
+	if err != nil {
+		return err
+	}
+	period := time.Duration(p)
```
Comment (benmoss) [inline pkg/eventshub/sender/sender.go:125]: Aren't `p` and `d` here already Durations?

File: pkg/eventshub/sender/sender.go [hunk 2]
```diff
@@ -117,8 +118,17 @@ func Start(ctx context.Context, logs *eventshub.EventLogs) error {
...
+
+	d, err := time.ParseDuration(durationWithUnit(env.Delay))
+	if err != nil {
+		return err
+	}
+	delay := time.Duration(d)
```
Comment (benmoss) [inline pkg/eventshub/sender/sender.go:131]:
```suggestion
	period, err := time.ParseDuration(durationWithUnit(env.Period))
	if err != nil {
		return err
	}

	delay, err := time.ParseDuration(durationWithUnit(env.Delay))
	if err != nil {
		return err
	}
```

Comment (lionelvillard) [reply pkg/eventshub/sender/sender.go]: duh! Thanks!

**Pull request merged**